tests/journald_tests.cpp

```diff
@@ -292,6 +293,189 @@ TEST_F(JournaldLoggerTest, ROOT_LogToJournaldWithBigLabel)
+  runState.forkedPid = pid;
+  executorState.runs.put(containerId, runState);
+
+  mesos::internal::slave::state::FrameworkState frameworkState;
+  frameworkState.id = executor.framework_id();
+  frameworkState.executors.put(executor.executor_id(), executorState);
+
+  mesos::internal::slave::state::SlaveState slaveState;
+  slaveState.id = state.id;
+  slaveState.frameworks.put(executor.framework_id(), frameworkState);
+
+  // Recover by using the mock `SlaveState`.
+  AWAIT_READY(containerizer->recover(slaveState));
+
+  status = containerizer->status(containerId);
+  AWAIT_READY(status);
+  ASSERT_TRUE(status->has_executor_pid());
+  EXPECT_EQ(pid, status->executor_pid());
+
+  // Now launch the nested container.
+  ContainerID nestedContainerId;
+  nestedContainerId.mutable_parent()->CopyFrom(containerId);
+  nestedContainerId.set_value(UUID::random().toString());
+
+  const std::string specialChildString = "special-child-string";
+
+  // This one can be short lived.
+  // We just need it to print a child-specific string and then exit.
+  launch = containerizer->launch(
+      nestedContainerId,
+      createContainerConfig(createCommandInfo(
+          "echo '" + specialChildString + "'")),
+      std::map<std::string, std::string>(),
+      None());
+
+  AWAIT_ASSERT_TRUE(launch);
+
+  status = containerizer->status(nestedContainerId);
+  AWAIT_READY(status);
+  ASSERT_TRUE(status->has_executor_pid());
+
+  // Wait for the nested container to finish.
+  Future<Option<mesos::slave::ContainerTermination>> nestedWait =
+    containerizer->wait(nestedContainerId);
+
+  AWAIT_READY(nestedWait);
+  ASSERT_SOME(nestedWait.get());
+
+  // Destroy the parent container.
+  Future<Option<mesos::slave::ContainerTermination>> wait =
+    containerizer->wait(containerId);
+
+  containerizer->destroy(containerId);
+
+  AWAIT_READY(wait);
+  ASSERT_SOME(wait.get());
+  ASSERT_TRUE(wait.get()->has_status());
+  EXPECT_WTERMSIG_EQ(SIGKILL, wait.get()->status());
+
+  // Now run two filters based on AGENT_ID and FRAMEWORK_ID.
+  // Normally, we would expect to get the same result from each filter.
+  // But in fact, we should *not* find the child string inside the
+  // query based on FRAMEWORK_ID because the MesosContainerizer
+  // will not persist that information after the emulated restart.
+  Future<std::string> firstQuery = runCommand(
+      "journalctl",
+      {"journalctl",
+      "AGENT_ID=" + stringify(state.id)});
+
+  Future<std::string> secondQuery = runCommand(
+      "journalctl",
+      {"journalctl",
+       "FRAMEWORK_ID=" + stringify(executor.framework_id())});
+
+  AWAIT_READY(firstQuery);
+  EXPECT_TRUE(strings::contains(firstQuery.get(), specialParentString));
+  EXPECT_TRUE(strings::contains(firstQuery.get(), specialChildString));
+
+  AWAIT_READY(secondQuery);
+  EXPECT_TRUE(strings::contains(secondQuery.get(), specialParentString));
+  EXPECT_FALSE(strings::contains(secondQuery.get(), specialChildString));
```

Comment:
So, the expectation is that this test would have passed even before your CHECK->if change, but that it will start to fail once we've "fixed" the bug in Mesos?